### PR TITLE
Add buildpackv3 build-template

### PR DESCRIPTION
**Pull request by jchesterpivotal**
This PR introduces preliminary support for [Cloud Native Buildpacks](https://buildpacks.io), aka buildpacks v3.

### CODE REVIEW

**Comment by jchesterpivotal** [comment]
I confirm for the record that @ekcasey, @dgodd and I are all covered by Pivotal's corporate CLA.

File: buildpack_v3/template.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    description: The run image buildpacks will use as the base for IMAGE.
+    default: packs/v3:run
+  - name: USE_CREDENTIAL_HELPERS
+    description: Use Docker credential helpers for Google's GCR, Amazon's ECR or Microsoft's ACR.
+    default: 'true'
+
+  steps:
+  - name: chown-home-workspace
+    image: packs/v3:latest
+    command: ["chown", "-R", "packs", "/builder/home", "/workspace"]
```
Comment (ImJasonH) [inline buildpack_v3/template.yaml]: Why is this necessary?

File: buildpack_v3/template.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  - name: copy-app
+    image: packs/v3:detect
+    command: ["cp", "-R", "/workspace", "/launch/app"]
```
Comment (ImJasonH) [inline buildpack_v3/template.yaml]: Can this be a symlink? Or can `packs` take an arbitrary source path?

File: buildpack_v3/template.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    volumeMounts:
+    - name: launch
+      mountPath: /launch
+    imagePullPolicy: Always
+  - name: detect
+    image: packs/v3:detect
+    workingDir: /lifecycleworkspace # Build creates a /workspace, which buildpacks tries to use for a different purpose.
```
Comment (ImJasonH) [inline buildpack_v3/template.yaml]: 🤦‍♂️

File: buildpack_v3/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+The Cloud Native Buildpacks website describes v3 buildpacks as:
+
+> ... pluggable, modular tools that translate source code into container-ready artifacts
+> such as OCI images. They replace Dockerfiles in the app development lifecycle with a higher level
+> of abstraction. ...  Cloud Native Buildpacks embrace modern container standards, such as the OCI
+> image format. They take advantage of the latest capabilities of these standards, such as remote
+> image layer rebasing on Docker API v2 registries.
+
+**Note**: This Buildpack v3 template is distinct from Buildpack.
```
[the remark this answers is not in the document]
Comment (sclevine) [reply buildpack_v3/README.md]: Agree with this. I'd like to see the original template modified to indicate that it provides the current CF buildpacks using the v2b API, and this template to provide links to the v3 stuff (e.g., buildpacks.io).

File: buildpack_v3/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+```
+kubectl apply -f https://raw.githubusercontent.com/knative/build-templates/master/buildpack_v3/template.yaml
+```
+
+## Parameters
+
+* **IMAGE:** The Docker image name to apply to the newly built image.
+    (_required_)
+* **RUN_IMAGE:** The run image buildpacks will use as the base for IMAGE.
+* **USE_CREDENTIAL_HELPERS:** Use Docker credential helpers. Set to `"true"` or `"false"` as string values.
```
[the remark this answers is not in the document]
Comment (sclevine) [reply buildpack_v3/README.md]: Also would prefer `USE_CRED_HELPERS` to match the convention elsewhere.

**Comment by sclevine** [comment]
https://github.com/buildpack/lifecycle/pull/14 should clean most of this up by letting us combine `/launch` and `/lifecycleworkspace` into just `/workspace` and removing the need to copy the app directory.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# Goland
```
Comment (sclevine) [inline .gitignore]: Not sure if it makes sense to include this file in the sample.

**Comment by sclevine** [comment]
@jchesterpivotal @dgodd took a stab at cleaning this up here: https://github.com/sclevine/build-templates/pull/1

File: .gitignore (hunk 6 above)
Comment (jchesterpivotal) [reply .gitignore]: It makes as much sense as any other gitignore, in my view. I think people will be more annoyed if I or someone else accidentally commits my `.idea` directory.

**Comment by ImJasonH** [comment]
I believe we're closing this in favor of #67
